source/common/mongo/proxy.cc

```diff
@@ -241,5 +261,51 @@ DecoderPtr ProdProxyFilter::createDecoder(DecoderCallbacks& callbacks) {
   return DecoderPtr{new DecoderImpl(callbacks)};
 }
 
+Optional<uint64_t> ProxyFilter::delayDuration() {
+  Optional<uint64_t> result;
+
+  if (!fault_config_) {
+    return result;
+  }
+
+  if (!runtime_.snapshot().featureEnabled(FIXED_DELAY_PERCENT_KEY, fault_config_->delayPercent())) {
+    return result;
+  }
+
+  uint64_t duration =
+      runtime_.snapshot().getInteger(FIXED_DELAY_DURATION_MS_KEY, fault_config_->delayDuration());
+
+  // Delay only if the duration is > 0ms.
+  if (duration > 0) {
+    result.value(duration);
+  }
+
+  return result;
+}
+
+void ProxyFilter::delayInjectionTimerCallback() {
+  delay_timer_.reset();
```

Comment:
Need to disable the timer.  Could make it a helper method.

Reply:
not necessary, at this point timer triggered already, so no need to disable.

Reply:
posted.